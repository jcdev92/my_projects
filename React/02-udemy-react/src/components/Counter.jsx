import React, {useState} from 'react';

const Counter = ({value}) => {

    const [count, setCount] = useState(value)

    const increment = () => (setCount(count + 1))
    const decrement = () => (setCount(count - 1))
    const reset = () => (setCount(value))

 return (
    <div className="counter__container">
        <h1>Counter</h1>
        <h2>{count}</h2>
        <button onClick={increment}>+</button>
        <button onClick={decrement}>-</button>
        <button aria-label="btn-reset" onClick={reset}>Reset</button>
    </div>
 )
};

export default Counter;