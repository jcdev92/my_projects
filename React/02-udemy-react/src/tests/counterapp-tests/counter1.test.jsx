import {fireEvent, screen, render} from "@testing-library/react";
import Counter from "../../components/Counter.jsx";

describe('testing de <Counter />', () => {
    const initialValue = 10;

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<Counter valueu={initialValue}/>)
        expect(container).toMatchSnapshot()
    })

    test('debe mostrar el valor por defecto de 10', () => {
        render(<Counter value={initialValue} />)
        // search for the element with the text 10 in the snapshot
            // expect(screen.getByText('10')).toBeTruthy()
        // searching the element by html tag level
            expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('10')
    })

    test('debe de incrementar con el boton +', () => {
        render(<Counter value={initialValue}/>)
        fireEvent.click(screen.getByText('+'))
        fireEvent.click(screen.getByText('+'))
        fireEvent.click(screen.getByText('+'))
        screen.debug()
        expect(screen.getByText('13')).toBeTruthy()

        fireEvent.click(screen.getByText('-'))
        fireEvent.click(screen.getByText('-'))
        screen.debug()
        expect(screen.getByText('11')).toBeTruthy()

        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        screen.debug()
        expect(screen.getByText('10')).toBeTruthy()

    })
})