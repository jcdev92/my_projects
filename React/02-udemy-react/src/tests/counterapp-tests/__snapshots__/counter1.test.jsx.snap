// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing de <Counter /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="counter__container"
  >
    <h1>
      Counter
    </h1>
    <h2 />
    <button>
      +
    </button>
    <button>
      -
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
